pkg/authn/kubernetes/keychain.go

```diff
@@ -270,12 +279,14 @@ func urlsMatch(globURL *url.URL, targetURL *url.URL) (bool, error) {
 	return true, nil
 }
 
-func toAuthenticator(configs []authn.AuthConfig) (authn.Authenticator, error) {
-	cfg := configs[0]
-
-	if cfg.Auth != "" {
-		cfg.Auth = ""
+func toAuthenticators(configs []authn.AuthConfig) []authn.Authenticator {
+	auths := make([]authn.Authenticator, 0, len(configs))
+	for _, cfg := range configs {
+		if cfg.Auth != "" {
```

Comment:
Should this also do the comparison against `empty` that we do in `keychain.go`? It may even be worth having an `isEmpty` helper (unexported) on `AuthConfig` if we find ourselves doing the comparison a lot.

Reply:
the checking is different, in `pkg/authn/keychain.go` is `if cfg != empty` but here only check `cfg.Auth`, maybe here has some reasons?